producer-kafka/src/main/java/io/stargate/producer/kafka/schema/SchemaRegistryProvider.java

```diff
@@ -0,0 +1,235 @@
+
+  private static final int SCHEMA_REGISTRY_MAX_CAPACITY = 1000;
+
+  private final SchemaRegistryClient schemaRegistryClient;
+
+  private MappingService mappingService;
+
+  public SchemaRegistryProvider(String schemaRegistryUrl, MappingService mappingService) {
+    this(
+        new CachedSchemaRegistryClient(schemaRegistryUrl, SCHEMA_REGISTRY_MAX_CAPACITY),
+        mappingService);
+  }
+
+  public SchemaRegistryProvider(
+      SchemaRegistryClient schemaRegistryClient, MappingService mappingService) {
+    this.schemaRegistryClient = schemaRegistryClient;
+    this.mappingService = mappingService;
+  }
+
+  @Override
+  public Schema getKeySchemaForTopic(String topicName) {
+    String subjectName = constructKeyRecordName(topicName);
+    return getLatestSchemaBySubject(subjectName);
+  }
+
+  @Override
+  public Schema getValueSchemaForTopic(String topicName) {
+    String subjectName = constructValueRecordName(topicName);
+    return getLatestSchemaBySubject(subjectName);
+  }
+
+  @Override
+  public void createOrUpdateSchema(TableMetadata tableMetadata) {
+    createOrUpdateKeySchema(tableMetadata);
+    createOrUpdateValueSchema(tableMetadata);
+  }
+
+  private Schema getLatestSchemaBySubject(String subjectName) {
+    int latestSchemaId = -1;
+    try {
+      // be careful - possible race condition, if between getLatestSchemaMetadata and
+      // getSchemaBySubjectAndId the createOrUpdateSchema is called
+      latestSchemaId = schemaRegistryClient.getLatestSchemaMetadata(subjectName).getId();
+      return (Schema)
+          schemaRegistryClient.getSchemaBySubjectAndId(subjectName, latestSchemaId).rawSchema();
+    } catch (IOException | RestClientException e) {
+      throw new RuntimeException(
+          "Problem when get key schema for subject: "
+              + subjectName
+              + " and schema id: "
+              + latestSchemaId,
+          e);
+    }
+  }
+
+  private void createOrUpdateValueSchema(TableMetadata tableMetadata) {
+    ParsedSchema valueSchema = new AvroSchema(constructValueSchema(tableMetadata));
+    String subject = constructValueRecordName(tableMetadata);
+
+    int schemaId = registerSchema(tableMetadata, valueSchema, subject);
+
+    logger.info(
+        "Registered valueSchema: {}, for subject: {} and id: {}", valueSchema, subject, schemaId);
+  }
+
+  private void createOrUpdateKeySchema(TableMetadata tableMetadata) {
+    ParsedSchema keySchema = new AvroSchema(constructKeySchema(tableMetadata));
+    String subject = constructKeyRecordName(tableMetadata);
+
+    int schemaId = registerSchema(tableMetadata, keySchema, subject);
+
+    logger.info(
+        "Registered keySchema: {}, for subject: {} and id: {}", keySchema, subject, schemaId);
+  }
+
+  private int registerSchema(TableMetadata tableMetadata, ParsedSchema schema, String subject) {
+    try {
+      return schemaRegistryClient.register(subject, schema);
```

Comment:
There are no async counterparts in the `SchemaRegistryClient`?

Reply:
No, there is only a sync version.
But on the other hand, we are wrapping calls to schema-registry within `CompletableFuture`, i.e.:
`return CompletableFuture.runAsync(() -> schemaProvider.createOrUpdateSchema(tableMetadata)); `
so there willl be no blocking from the perspective of the stargate core CDC framework